src/ansys/fluent/core/utils/file_transfer_service.py

```diff
@@ -229,7 +234,12 @@ def upload(
         files = _get_files(file_name, self.host_mount_path)
         if self.client:
             for file in files:
-                if os.path.isfile(file):
+                is_file_exists_on_remote = self.file_exists_on_remote(
+                    os.path.basename(file)
+                )
+                if is_file_exists_on_remote:
+                    print(f"\n{file} already uploaded.\n")
+                elif os.path.isfile(file) and not is_file_exists_on_remote:
```

Comment:
According to my understanding, this aligns the gRPC file transfer service behaviour with the PIM based behaviour that's (currently) implemented in a separate class. From that point of view, we can merge this now, and then consider what the overall behaviour should be and to what extent it should be configurable by the user.

Reply:
Yes, sure.